backend/src/db/migrations/20250718133527_project-unify-revert.ts

```diff
@@ -0,0 +1,342 @@
+      projectSlackConfigs.map((el) => {
+        const id = uuidV4();
+        return { ...el, id, projectId: newProjectId };
+      })
+    );
+  }
+
+  const projectMicrosoftTeamsConfigs = await knex(TableName.ProjectMicrosoftTeamsConfigs).where("projectId", projectId);
+  if (projectMicrosoftTeamsConfigs.length) {
+    await knex.batchInsert(
+      TableName.ProjectMicrosoftTeamsConfigs,
+      projectMicrosoftTeamsConfigs.map((el) => {
+        const id = uuidV4();
+        return { ...el, id, projectId: newProjectId };
+      })
+    );
+  }
+
+  const trustedIps = await knex(TableName.TrustedIps).where("projectId", projectId);
+  if (trustedIps.length) {
+    await knex.batchInsert(
+      TableName.TrustedIps,
+      trustedIps.map((el) => {
+        const id = uuidV4();
+        return { ...el, id, projectId: newProjectId };
+      })
+    );
+  }
+
+  return newProjectId;
+};
+
+const kickOutSecretManagerProject = async (knex: Knex, oldProjectId: string) => {
+  const secret = await knex(TableName.Secret)
+    .join(TableName.SecretFolder, `${TableName.SecretFolder}.id`, `${TableName.Secret}.folderId`)
+    .join(TableName.Environment, `${TableName.Environment}.id`, `${TableName.SecretFolder}.envId`)
+    .where("projectId", oldProjectId)
+    .returning(`${TableName.Secret}.id`)
+    .first();
+  if (secret) {
+    const newProjectId = await newProject(knex, oldProjectId, ProjectType.SecretManager);
+    await knex(TableName.IntegrationAuth).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.SecretBlindIndex).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.SecretSync).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.SecretTag).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.SecretReminderRecipients).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.ServiceToken).where("projectId", oldProjectId).update("projectId", newProjectId);
+  }
+};
+
+const kickOutCertManagerProject = async (knex: Knex, oldProjectId: string) => {
+  const cas = await knex(TableName.CertificateAuthority).where("projectId", oldProjectId).returning("id").first();
+  if (cas) {
+    const newProjectId = await newProject(knex, oldProjectId, ProjectType.CertificateManager);
+    await knex(TableName.CertificateAuthority).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.Certificate).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.PkiSubscriber).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.PkiCollection).where("projectId", oldProjectId).update("projectId", newProjectId);
+    await knex(TableName.PkiAlert).where("projectId", oldProjectId).update("projectId", newProjectId);
+  }
```

Comment:
it might be better here to check for both Certificate and CertificateAuthority table